pylint/checkers/typecheck.py

```diff
@@ -1058,6 +1064,39 @@ def _check_uninferable_call(self, node):
                     )
                     break
 
+    def _check_argument_order(self, node, call_site, called, called_param_names):
+        """Match the supplied argument names against the function parameters.
+        Warn if some argument names are not in the same order as they are in
+        the function signature.
+        """
+        # Check for called function being an object instance function
+        # If so, ignore the initial 'self' argument in the signature
+        try:
+            if (isinstance(called.parent, astroid.scoped_nodes.ClassDef)
+                and called_param_names[0] == 'self'):
+                called_param_names = called_param_names[1:]
+        except IndexError:
+            return
+
+        # extract argument names, if they have names
+        try:
+            calling_arg_names = [p.name for p in call_site.positional_arguments]
+        except AttributeError:
+            # the type of arg does not provide a `.name`. In this case we
+            # stop checking for out-of-order arguments because it is only relevant
+            # for named variables.
+            return
+
+        # Only warn if all names match something in called function params
```

Comment:
All the calling argument names should match parameter names, but on the other hand, every parameter name should be matched by a calling argument name. I think that's why there are false positives where the same argument appears in multiple positions.

Now, maybe we want to say that if there is a function defined as `f(x, y)` which is then called as `f(x, x)`, that's worth warning about. Maybe it would be better to write `f(x, y=x)` to make it clear that it's intentional. That's probably a good idea, but users might find that overbearing and heavy-handed. When it comes to linters, people get really pissy when they encounter a false positive or anything that feels like a false positive.

Reply:
Sounds reasonable and I agree with erring on the side of false positives.

I have altered to check to only emit on a full symmetric match between the set of arguments and the set of parameters.